packages/shell-api/src/collection.spec.ts

```diff
@@ -1170,4 +1175,115 @@ describe('Collection', () => {
+      initializeUnorderedBulkOp: { m: 'initializeBulkOp' },
+      distinct: { i: 4 },
+      estimatedDocumentCount: { i: 2 },
+      findAndModify: { i: 5 },
+      findOneAndReplace: { i: 4 },
+      findOneAndUpdate: { i: 4 },
+      replaceOne: { i: 4 },
+      updateMany: { i: 4 },
+      updateOne: { i: 4 },
+      getIndexes: { i: 2 },
+      reIndex: { i: 2 },
+    };
+    const ignore = [ 'getShardDistribution', 'stats', 'isCapped' ];
+    const args = [ { query: {} }, {}, { out: 'coll' } ];
+    beforeEach(() => {
+      const bus = stubInterface<EventEmitter>();
+      serviceProvider = stubInterface<ServiceProvider>();
+      serviceProvider.initialDb = 'test';
+      serviceProvider.bsonLibrary = bson;
+      internalSession = stubInterface<ServiceProviderSession>();
+      serviceProvider.startSession.returns(internalSession);
+      serviceProvider.aggregate.returns(stubInterface<ServiceProviderCursor>());
+      serviceProvider.find.returns(stubInterface<ServiceProviderCursor>());
+      serviceProvider.getIndexes.resolves([]);
+      serviceProvider.stats.resolves({ storageSize: 1, totalIndexSize: 1 });
+      serviceProvider.listCollections.resolves([]);
+      serviceProvider.countDocuments.resolves(1);
+
+      serviceProvider.runCommandWithCheck.resolves({ ok: 1, version: 1, bits: 1, commands: 1, users: [], roles: [], logComponentVerbosity: 1 });
+      [ 'bulkWrite', 'deleteMany', 'deleteOne', 'insert', 'insertMany',
+        'insertOne', 'replaceOne', 'update', 'updateOne', 'updateMany',
+        'findOneAndDelete', 'findOneAndReplace', 'findOneAndUpdate',
+        'findAndModify'
+      ].forEach(
+        k => serviceProvider[k].resolves({ result: {}, value: {} })
+      );
+      const internalState = new ShellInternalState(serviceProvider, bus);
+      const mongo = new Mongo(internalState);
+      const session = mongo.startSession();
+      collection = session.getDatabase('db1').getCollection('coll');
+    });
+    it('all commands that use the same command in sp', async() => {
+      for (const method of Object.getOwnPropertyNames(Collection.prototype).filter(
+        k => !ignore.includes(k) && !Object.keys(exceptions).includes(k)
+      )) {
+        if (!method.startsWith('_') &&
+          !method.startsWith('print') &&
+          collection[method].returnsPromise) {
+          try {
+            await collection[method](...args);
+          } catch (e) {
+            expect.fail(`${method} failed, error thrown ${e.message}`);
+          }
+          expect(serviceProvider[method].calledOnce).to.equal(true, `expected ${method} to be called but it was not`);
+          expect((serviceProvider[method].getCall(-1).args[3] as any).session).to.equal(internalSession);
+        }
+      }
+    });
+    it('all commands that use other methods', async() => {
+      for (const method of Object.getOwnPropertyNames(Collection.prototype).filter(
+        k => Object.keys(exceptions).includes(k)
+      )) {
```

Comment:
Do we just want to
```suggestion
      for (const method of Object.keys(exceptions)) {
```

(and ditto for `database`)? It would be unexpected for any key of the exceptions object to be missing as a Collection method, right?

Reply:
Most of the database methods are actually _not_ present in `exceptions`. The methods present in `exceptions` are the methods that behave differently from the standard methods, for example if they have their options argument in a different location or they don't use runCommand and instead use some other driver method.

I also want to apply this test to _all_ database and collection methods, including future ones.

Reply:
Oh oops nevermind, I totally misread the suggestion!